### DISCUSSION TITLE
Poor RGB Camera Images with Resolution 1920x1080, cloudy=0

### DISCUSSION THREAD

**Discussion by yyhtbs-yye** [carla-simulator/carla#6529] [Q&A]
<img width="455" alt="image" src="https://github.com/carla-simulator/carla/assets/74354145/f666800c-0e88-4833-a4cc-4ee341aa5fc4">
<img width="351" alt="image" src="https://github.com/carla-simulator/carla/assets/74354145/5b96719f-9eda-4436-af25-bca1d4de77f6">

I am current using Carla to capture road traffic, however, the rgb camera image seems very noisy, and somtimes the car frame seems transparent, as shown above. After checking the issues, I found the reason might be relate to 

[ref_sensors](https://carla.readthedocs.io/en/latest/ref_sensors/)
> If enable_postprocess_effects is enabled, a set of post-process effects is applied to the image for the sake of realism:

> Vignette: darkens the border of the screen.
> **Grain jitter**: adds some noise to the render.
> Bloom: intense lights burn the area around them.
> Auto exposure: modifies the image gamma to simulate the eye adaptation to darker or brighter areas.
> Lens flares: simulates the reflection of bright objects on the lens.
> Depth of field: blurs objects near or very far away of the camera.

Maybe the gitter causes the noises? 

Is it possible to disable some postprocessing features in the latest release?

Appreciate for any helps!

Best Regards
Yuhang

**Comment by yyhtbs-yye** [comment]
This is the case when the postprocess effects are disable and Gamma is somehow corrected. 
<img width="476" alt="image" src="https://github.com/carla-simulator/carla/assets/74354145/0e0ce6c4-8bb3-4f3e-8288-ba98be922013">

**Reply by yyhtbs-yye** [reply]
The jitter is obviously a problem, and might be better disabled for the postprocessing.

**Reply by xfqbuaa** [reply]
only remind you to first check nvidia GPU works or not?
To run CARLA with nvidia driver 
./CarlaUE4.sh -prefernvidia

**Reply by yyhtbs-yye** [reply]
I use 3060 laptop, when game runs, gou works, I can confirm that from nvidia-smi. The game window quality is high but when it comes to the RGB camera image, the quality becomes quite low

**Comment by yyhtbs-yye** [comment]
The problem solved by recompiling the carla source file. 
The part to change is "YOUR_CALAR_ROOT_FOLDER\Unreal\CarlaUE4\Plugins\Carla\Source\Carla\Sensor\SceneCaptureSensor.cpp"

The line to change is 704 -> follow a few lines, the changed code are given below, 

<img width="680" alt="image" src="https://github.com/carla-simulator/carla/assets/74354145/c8f8b4df-6ba8-4226-9e27-f9ba950c29bc">

I am trying to disable some of the features, the main jitter noise is the "ShowFlags.AntiAliasing", set it to false to remove the jitter. 

I have a following question is that, can the compiling be faster? it seems like always compiling from the very begining (might be wrong). The code I modified is just a plugin, it should not be recompiled from very begining. It may cause difficulties to debug.

**Reply by alexx-km** [reply]
I have the same issue, but the settings you suggested don't seem to fix the poor quality. Have you tried any other settings or modifications in the meantime? 
Thank you in advance!

**Reply by alexx-km** [reply]
@yyhtbs-yye Any updates on this topic? I'm still facing the issue and rebuilding has not worked for me.

**Reply by yyhtbs-yye** [reply]
Try out different combinations. It should be of good performance disabling some. For details, I cannot really remeber.

**Reply by yyhtbs-yye** [reply]
Difussor might be the problem

**Reply by alexx-km** [reply]
Thank you for your reply! I tried several combinations (with Anti-Aliasing cause without it looked even worse) and the cars in the distance still seem to be grainy / fuzzy: 
![RGB-CAM_BACK_1706374685750](https://github.com/carla-simulator/carla/assets/60221685/38ada5be-d4d0-4b26-bd4c-0087ffdec040)
![image](https://github.com/carla-simulator/carla/assets/60221685/74f4c9f6-204a-4d36-b754-ce8981ef62de)
Or is this the maximum quality that you can expect from the simulator?

**Reply by yyhtbs-yye** [reply]
This might be the best quality alreay already. For the grainy stuff, maybe check if there are some other anti-aliasing algorithms for image (not low pass filtering) available as post processing.

**Reply by alexx-km** [reply]
That is a pity. I tried several other things (e.g. using a higher resolution for the cameras and downsizing with OpenCV) but this seems to be the best quality possible. Using post-processing algorithms would be an option, but then generating datasets would take even more time. I could swear that with older versions of CARLA (pre 0.9.13) the quality was better...

**Comment by yyhtbs-yye** [comment]
> Thank you for your reply! I tried several combinations (with Anti-Aliasing cause without it looked even worse) and the cars in the distance still seem to be grainy / fuzzy: ![RGB-CAM_BACK_1706374685750](https://private-user-images.githubusercontent.com/60221685/300254820-38ada5be-d4d0-4b26-bd4c-0087ffdec040.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY1MjE5NTUsIm5iZiI6MTcwNjUyMTY1NSwicGF0aCI6Ii82MDIyMTY4NS8zMDAyNTQ4MjAtMzhhZGE1YmUtZDRkMC00YjI2LWJkNGMtMDA4N2ZmZGVjMDQwLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTI5VDA5NDczNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY4ZWEwYzBmMjJhZjlhODViOGI2NzMxMjdlYmY0NWM3YzFkNmNjZTQwYzA0OTBkMWZkMDA3YjNjMTk1MWYzOTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.8feUbbY5hG6OdIAqMWWZcOWbsT98sCs8gtOgwMkZzjo) ![image](https://private-user-images.githubusercontent.com/60221685/300255121-74f4c9f6-204a-4d36-b754-ce8981ef62de.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY1MjE5NTUsIm5iZiI6MTcwNjUyMTY1NSwicGF0aCI6Ii82MDIyMTY4NS8zMDAyNTUxMjEtNzRmNGM5ZjYtMjA0YS00ZDM2LWI3NTQtY2U4OTgxZWY2MmRlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTI5VDA5NDczNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU3ZDJiNmQ1NmE4ZWZkNTBlYTc4NzI2OTgxZmMwZjRmOGZiOTY0NGY0MzkwZTkwOTdhZTY4MDI5NGY0NDU3MGUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.MrJhtCrf0nzlHV3fVy3AmIq5T44uTHxc4rq0HlwStsg) Or is this the maximum quality that you can expect from the simulator?

**Comment by luke3322** [comment]
I'm facing the same issue. Is that a problem everyone encounters when setting up RGB cameras or is something special trigging the effect depending on system/hardware/setup? And can you only "solve" it by recompiling it with the changes you mentioned? Because myself using just the win NoEditor version.

**Reply by alexx-km** [reply]
I was not able to solve it using the changes mentioned. I also tested it on different hardware setups (GTX Titan X / RTX 3060) on Windows and Ubuntu.

**Comment by LuisPoveda** [comment]
Thank you for posting this issue, we have created an issue to look into it, we will post here when we have some updates.
We are sorry for the delay in response, we will do our best to reduce response times.

**Reply by YonatanSimson** [reply]
@LuisPovedaCano Have you looked into this issue? I have the same problem with 9.15 running in windows on a precompiled version. GPU: RTX 4060
